.github/workflows/py-unit-tests.yml

```diff
@@ -1,52 +1,18 @@
-# Workflow that runs frontend and python unit tests
-name: Run Unit Tests
-
-# Only run one workflow of the same group at a time.
-# There can be at most one running and one pending job in a concurrency group at any time.
-concurrency:
```

Comment:
Why remove these?

Reply:
Concurrency is used when two workflows/jobs conflict with each other and you don't want them to run at the same time. For example our Deploy Doc job, if two run at the same time, it will fail one of them.
Unit tests have no concurrency issues and can always run without overlap.

Reply:
If we can avoid unneccessary runs of tests, imo we should keep the concurrency options in place.

Reply:
Concurrency doesn't solve that issue from what I understand. 
It will still re-run tests. It's just a matter of not running them at the same time.